### ISSUE TITLE
Collection dump requires namespace to be in oplog

### ISSUE THREAD

**Issue by Perchework** [mongodb-labs/mongo-connector#310]
Hi,

I found a problem after restoring all my collections in mongodb.
I've got really big collections and after the restore, only the last collection stay in oplog because of the oplog size limit.
The collection wich is in oplog is named a.b.

If I start mongo-connector with -n a.b, initial collection dump work as expected :
```
mongo_connector.oplog_manager:153 - OplogThread: Run thread started
mongo_connector.oplog_manager:155 - OplogThread: Getting cursor
mongo_connector.oplog_manager:661 - OplogThread: reading last checkpoint as None
mongo_connector.oplog_manager:395 - OplogThread: Dumping set of collections ['a.b']
mongo_connector.oplog_manager:573 - OplogThread: Last oplog entry has timestamp 1440088019.
mongo_connector.oplog_manager:493 - OplogThread: Using bulk upsert function for collection dump
```

But if I start mongo-connector with -n a.c ({ns: a.c} cannot be found in oplog), there is no collection dump:
```
mongo_connector.oplog_manager:153 - OplogThread: Run thread started
mongo_connector.oplog_manager:155 - OplogThread: Getting cursor
mongo_connector.oplog_manager:661 - OplogThread: reading last checkpoint as None
mongo_connector.oplog_manager:395 - OplogThread: Dumping set of collections ['a.c']
mongo_connector.oplog_manager:167 - OplogThread: Last entry is the one we already processed.  Up to date.  Sleeping.
```

If I update a document in mongodb on collection a.c, {ns: a.c} can be found in oplog and the collection dump is possible... but i can't update a document in each collections after restoring all my mongo collection to have mongo-connector working...

Thank for your help,
And Sorry for my really bad english

Thomas.

**Comment by Perchework** [comment]
the problem come from oplog_manager.py l 569 :
```python
if curr.count(with_limit_and_skip=True) == 0:
    return None
```
the function count send this query to mongo 
```
db.getCollection('oplog.rs').find({ ns: { $in: [ "a.c", "a.$cmd", "admin.$cmd" ] } })
```
and return nothing because i've got only {ns: a.b} in db.getCollection('oplog.rs') ...

Thanks

**Comment by sbrunov** [comment]
Hello!

Have faced the same problem with the support of the `namespaces.include` configuration parameter after backing up and restoring the database without the oplog. It would be great to have the appropriate support of the parameter.

I would like to propose the change (below) as a workaround (because it's not optimal?) while the limitation:
> <…> So, trying to use mongo-connector **specifically on one collection wouldn't sync anything** because there are no entries for that collection. <…>
>
> [mongodb-labs/mongo-connector, FAQ](https://github.com/mongodb-labs/mongo-connector/wiki/FAQ#why-is-the-last-entry-already-processed--up-to-date-while-using-namespace-command-line-args-even-though-collections-are-not-synced-to-destination).

is present.

The proposed "patch" for the `mongo_connector/oplog_manager.py` file:

```python
    def run(self):

                         #sync the current oplog operation
                         operation = entry['op']
                         ns = entry['ns']
 
                         if '.' not in ns:
                             continue
                         coll = ns.split('.', 1)[1]
 
+                        # Ignore the collection if it is not included
+                        if ns not in self.oplog_ns_set:
+                            LOG.debug("OplogThread: Skip the collection %s: it is not included." % ns)
+                            continue
+
                         # Ignore system collections
                         if coll.startswith("system."):
                             continue
```

```python
     def get_oplog_cursor(self, timestamp=None):
         """Get a cursor to the oplog after the given timestamp, filtering
         entries not in the namespace set.
         If no timestamp is specified, returns a cursor to the entire oplog.
         """
         query = {}
-        if self.oplog_ns_set:
-            query['ns'] = {'$in': self.oplog_ns_set}
```


```python
     def get_last_oplog_timestamp(self):
         """Return the timestamp of the latest entry in the oplog.
         """
-        if not self.oplog_ns_set:
-            curr = self.oplog.find().sort(
-                '$natural', pymongo.DESCENDING
-            ).limit(1)
-        else:
-            curr = self.oplog.find(
-                {'ns': {'$in': self.oplog_ns_set}}
-            ).sort('$natural', pymongo.DESCENDING).limit(1)
+        curr = self.oplog.find().sort(
+            '$natural', pymongo.DESCENDING
+        ).limit(1)
```

It seems to work just fine!

Any comments and suggestions would be appreciated.

Thank you!

Best regards,
Sergey Brunov.

**Comment by llvtt** [comment]
Hi @sbrunov! I think your approach is the necessary one to solve this problem. Would you like to make a pull request for this issue? Thanks!

**Comment by sbrunov** [comment]
Hello, @llvtt!

Thank you for your confirmation.
Have just created the pull request.

Best regards,
Sergey Brunov.

**Comment by aherlihy** [comment]
@sbrunov Hello! Thank you for your work. Unfortunately, it seems your [pull request](https://github.com/mongodb-labs/mongo-connector/pull/367) did not pass the [test suite](https://travis-ci.org/mongodb-labs/mongo-connector/builds/95393179). Are you still working on the feature, or do you have any updates? Thanks!

**Comment by gisaia** [comment]
Dear @sbrunov,
Did you succeed in integrating your patch in the mongo-connector? If so, is it in the master branch?
I am having troubles while trying to sync several collections in various ES indices by running one connector process for each collection (with -n and -g options) : one collection has been synched while the rest is not. It looks like the issue you raised.
Thank you for your help.

**Comment by ShaneHarvey** [comment]
Hi @Perchework, @gisaia, and @sbrunov,

The fix for this bug (https://github.com/mongodb-labs/mongo-connector/commit/2b48c95cb3c7a118c939040e1781cff0677b5019) is included in the 2.4.1 release of mongo-connector. If you are still facing problems after upgrading please open a new issue. 

Thanks!
